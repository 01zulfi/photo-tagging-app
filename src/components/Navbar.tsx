import React, { FC } from 'react';
import styled from 'styled-components';
import Timer from './Timer';

interface NavbarProps {
  shouldTimerStart: boolean;
}

const StyledNavbar = styled.header`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.05);
  padding-left: 20px;
  padding-right: 20px;
`;

const StyledHeading = styled.h1`
  @import url('https://fonts.googleapis.com/css2?family=Anton&display=swap');
  font-family: 'Anton', sans-serif;
  color: white;
`;

const StyledLBButton = styled.button`
  height: fit-content;
  padding: 20px;
  background-color: #bb86fc;
  border: 0;
  border-radius: 10px;
  &:hover {
    color: white;
    background: #651fff;
    cursor: pointer;
  }
`;

const Zero = styled.p`
  @import url('https://fonts.googleapis.com/css2?family=Anton&display=swap');
  font-family: 'Anton', 'sans-serif';
  color: white;
  font-size: 2rem;
  background: rgba(255, 255, 255, 0.07);
  border-radius: 10px;
  width: 25%;
  text-align: center;
  margin-left: auto;
  margin-right: 5%;
`;

const Navbar: FC<NavbarProps> = ({ shouldTimerStart }) => (
  <StyledNavbar>
    <StyledHeading>find the characters</StyledHeading>
    {shouldTimerStart ? <Timer /> : <Zero>0s</Zero>}
    <StyledLBButton>Leaderboard</StyledLBButton>
  </StyledNavbar>
);
export default Navbar;
