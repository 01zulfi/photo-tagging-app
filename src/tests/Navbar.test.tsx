import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

test('matches snapshot', () => {
  expect(render(<Navbar shouldTimerStart={false} />)).toMatchSnapshot();
});

test('renders correct heading', () => {
  const { getByRole } = render(<Navbar shouldTimerStart={false} />);
  expect(getByRole('heading')).toHaveTextContent(/find the characters/i);
});
