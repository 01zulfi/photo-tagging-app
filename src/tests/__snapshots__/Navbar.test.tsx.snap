// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-gsDKAQ kXebAn"
      >
        <h1
          class="sc-dkPtRN dgEvov"
        >
          find the characters
        </h1>
        <p
          class="sc-eCImPb dHgcqA"
        >
          0s
        </p>
        <button
          class="sc-hKwDye emkORy"
        >
          Leaderboard
        </button>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-gsDKAQ kXebAn"
    >
      <h1
        class="sc-dkPtRN dgEvov"
      >
        find the characters
      </h1>
      <p
        class="sc-eCImPb dHgcqA"
      >
        0s
      </p>
      <button
        class="sc-hKwDye emkORy"
      >
        Leaderboard
      </button>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
